ChatLayout/Classes/Core/CollectionViewChatLayout.swift

```diff
@@ -619,7 +627,7 @@ open class CollectionViewChatLayout: UICollectionViewLayout {
             || (isUserInitiatedScrolling && state == .beforeUpdate)
 
         invalidationActions.remove(.shouldInvalidateOnBoundsChange)
-        return shouldInvalidateLayout
+        return shouldInvalidateLayout || hasPinnedHeaderOrFooter
```

Comment:
Update the pinned header and footer correctly when the offset changes.

When using pinToVisibleBounds with UICollectionViewCompositionalLayout, it always returns true.